### PR TITLE
feat(sdk): add EntityIdentifier convenience constructors

### PR DESCRIPTION
**Pull request by marythought**
## Summary
- Add `forEmail`, `forClientId`, `forUserName`, `forToken`, and `withRequestToken` helper functions that mirror the Go SDK's EntityIdentifier constructors
- Reduces authorization call setup from ~15 lines of nested object literals to a single function call
- Exported from `@opentdf/sdk` top-level

**Before:**
```typescript
const response = await platform.v2.authorization.getDecision({
  entityIdentifier: {
    identifier: {
      case: 'entityChain',
      value: {
        entities: [
          {
            entityType: {
              case: 'emailAddress',
              value: 'jen@example.com',
            },
            category: Entity_Category.SUBJECT,
          },
        ],
      },
    },
  },
  action: { name: 'read' },
  resource: {
    resource: {
      case: 'attributeValues',
      value: {
        fqns: ['https://example.com/attr/department/value/finance'],
      },
    },
  },
});
```

**After:**
```typescript
import { forEmail } from '@opentdf/sdk';

const response = await platform.v2.authorization.getDecision({
  entityIdentifier: forEmail('jen@example.com'),
  action: { name: 'read' },
  resource: {
    resource: {
      case: 'attributeValues',
      value: {
        fqns: ['https://example.com/attr/department/value/finance'],
      },
    },
  },
});
```

## Test plan
- [x] 9 mocha tests covering all 5 helpers with valid and empty string inputs
- [x] Prettier + ESLint clean

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **New Features**
  * Added helper functions for constructing authorization entity identifiers. Enables identifying entities by email address, client ID, username, JWT token, or automatic extraction of tokens from request authorization headers for simplified implementation and seamless integration.

* **Tests**
  * Added comprehensive unit tests for entity identifier helper functions, including validation of all identifier types and edge cases.

### CODE REVIEW

File: lib/src/platform/authorization/entity-identifiers.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+/** Returns an EntityIdentifier that resolves the entity from the given JWT. */
+export function forToken(jwt: string): EntityIdentifier {
+  return create(EntityIdentifierSchema, {
+    identifier: {
+      case: 'token',
+      value: create(TokenSchema, { jwt }),
+    },
+  });
+}
```
[the remark this answers is not in the document]
Comment (marythought) [reply lib/src/platform/authorization/entity-identifiers.ts:80]: The Go SDK doesn't have a `ForRegisteredResourceValueFQN` helper — the helpers in this PR match the Go SDK's actual surface. Happy to add it in a follow-up if there's demand.

File: lib/src/platform/authorization/entity-identifiers.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+function fromEntity(entity: Entity): EntityIdentifier {
+  return create(EntityIdentifierSchema, {
+    identifier: {
+      case: 'entityChain',
+      value: create(EntityChainSchema, { entities: [entity] }),
+    },
+  });
```
[the remark this answers is not in the document]
Comment (marythought) [reply lib/src/platform/authorization/entity-identifiers.ts:101]: The Go SDK's `entityIdentifierFromEntity` is private (unexported, lowercase). Keeping `fromEntity` private here matches that design — the public helpers cover the common cases and keep the API surface small.

**Review by eugenioenko — APPROVED** [review]
🎉
